source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -713,13 +715,20 @@ def call_kraken(self, request_type, request, instance, krakens_call):
         # TODO: call first bss|bss and do not call walking|walking if no bss in first results
         resp = []
         logger = logging.getLogger(__name__)
+        futures = {}
         for dep_mode, arr_mode in krakens_call:
             pb_request = create_pb_request(request_type, request, dep_mode, arr_mode)
-            self.nb_kraken_calls += 1
 
-            local_resp = instance.send_and_receive(pb_request)
+            #we spawn a new green thread, it won't have access to our thread local request object so we set request_id
+            futures[(dep_mode, arr_mode)] = gevent.spawn(instance.send_and_receive,
+                                                         pb_request,
+                                                         request_id=flask.request.id)
 
+        gevent.joinall(futures.values())
+        for (dep_mode, arr_mode), future in futures.items():
```

Comment:
You can just remove the joinall. `future.get()` will wait. That's not as optimal as `iwait()`, but better than `joinall()`.

Or you can spawn a closure returning `(dep_mode, arr_mode, instance.send_and_receive(...))`